funtofem/driver/funtofem_nlbgs_driver.py

```diff
@@ -157,6 +160,22 @@ def _solve_steady_forward(self, scenario, steps=None):
                 body.transfer_loads(scenario)
                 body.transfer_heat_flux(scenario)
 
+                if self._debug:
```

Comment:
You need a self.comm.rank == 0 check here so for large proc cases you don't print this out 48 times, etc. If you do want it to print out for every processor then we should have a different debug mode like `debug = "all"`, `debug = "root"` and `debug=None` or something.
```suggestion
                if self._debug and self.comm.rank == 0:
```